specs/BrowserProcessExited.md

```diff
@@ -66,19 +78,33 @@ void AppWindow::CloseWebView(/* ... */) {
               ICoreWebView2Environment* sender,
               ICoreWebView2BrowserProcessExitedEventArgs* args) {
           COREWEBVIEW2_BROWSER_PROCESS_EXIT_KIND kind;
+          UINT32 pid;
           CHECK_FAILURE(args->get_BrowserProcessExitKind(&kind));
-
-          // Watch for graceful browser process exit. Let ProcessFailed event
-          // handler take care of failed browser process termination.
-          if (kind == COREWEBVIEW2_BROWSER_PROCESS_EXIT_KIND_NORMAL)
+          CHECK_FAILURE(args->get_BrowserProcessId(&pid));
+
+          // If a new WebView is created from this CoreWebView2Environment after
+          // the browser has exited but before our handler gets to run, a new
+          // browser process will be created and lock the user data folder
+          // again. Do not attempt to cleanup the user data folder in these
+          // cases. We check the PID of the exited browser process against the
+          // PID of the browser process to which our last CoreWebView2 attached.
+          if (pid == m_newestBrowserPid)
           {
-            CHECK_FAILURE(
-                m_webViewEnvironment->remove_BrowserProcessExited(m_browserExitedEventToken));
-            // Release the environment only after the handler is invoked.
-            // Otherwise, there will be no environment to raise the event when
-            // the collection of WebView2 Runtime processes exit.
-            m_webViewEnvironment = nullptr;
-            CleanupUserDataFolder();
+            // Watch for graceful browser process exit. Let ProcessFailed event
+            // handler take care of failed browser process termination.
+            if (kind == COREWEBVIEW2_BROWSER_PROCESS_EXIT_KIND_NORMAL)
+            {
+              CHECK_FAILURE(
+                  m_webViewEnvironment->remove_BrowserProcessExited(m_browserExitedEventToken));
+              // Release the environment only after the handler is invoked.
+              // Otherwise, there will be no environment to raise the event when
+              // the collection of WebView2 Runtime processes exit.
+              m_webViewEnvironment = nullptr;
+              CleanupUserDataFolder();
+            }
+          } else {
+            MessageBox(m_mainWindow, L"A new browser process prevented cleanup of "
+                            L"the user data folder.", L"Cleanup User Data Folder", MB_OK);
```

Comment:
Is there something we recommend end devs do in this situation? Or its just an error and they should communicate between their app instances to not lock the user data folder?

Reply:
It's just an error. The app shouldn't be trying to do cleanup while another process has a webview open over the same user data folder.